[6.0] Integration of category change when transition in workflow

Dileepadari opened the pull request:
Pull Request for Issue # .

### Summary of Changes


### Testing Instructions



### Actual result BEFORE applying this Pull Request



### Expected result AFTER applying this Pull Request



### Link to documentations
Please select:
- [ ] Documentation link for docs.joomla.org: <link>
- [ ] No documentation changes for docs.joomla.org needed

- [ ] Pull Request link for manual.joomla.org: <link>
- [ ] No documentation changes for manual.joomla.org needed

## Summary by Sourcery

Introduce a new workflow plugin that lets administrators specify and apply category changes when transitioning articles, enforcing the category option and updating articles accordingly

New Features:
- Add Workflow Category plugin to assign categories during content workflow transitions
- Provide a category selection field in the workflow transition form

Enhancements:
- Make the article form and batch category selector read-only to enforce transition-driven changes

Chores:
- Register the plugin via a DI service provider and include its manifest files
- Include language files for UI labels and status messages

bembelimen committed:
Release Joomla! 5.3.1

Dileepadari committed:
added category transition plugin

Dileepadari committed:
added category transition plugin

Dileepadari committed:
added category transition plugin

brianteeman committed:
[5.4] Automated Updates information (#45517)

When using a local dev environment or site that is not publicly accessible such as an intranet the quickicon will display "automated updates connection broken"

If you then click on the quickicon you will still not see why it is broken

If you then try and save the options only then do you get the technical reason
danger
Error while registering to automated update service: cURL error 6: Could not resolve host: j51.test (see https://curl.haxx.se/libcurl/c/libcurl-errors.html) for https://j51.test/api/index.php/v1/joomlaupdate/healthcheck (500).

This PR adds a fieldset description (displayed as information) which is always displayed

Signed-off-by: BrianTeeman <brian@teeman.net>

Bodge-IT committed:
Merge remote-tracking branch 'upstream/5.4-dev' into upmerges/2025-05-25

Bodge-IT committed:
[6.0] Upmerges - 2025-05-25

Merge pull request #45526 from Bodge-IT/upmerges/2025-05-25
Pre -alpha prep.

richard67 committed:
Add the compat6 plugins with enabled options (#45525)

muhme committed:
Joomla! 5.4.0 Alpha 1

muhme committed:
Revert to dev

HLeithner committed:
[6.0] Introduce Joomla 6.0 compatibility plugin (#45480)

* Create backward compatibility 6 plugin

* Moved relevant code from old compat plugin and removed it

considered changes from prs:
#44926
#44240
#44638
#44700
#43209
#43958
#42884

* Add new plugin on install and uninstall old plugin

* Remove compat media folder, add compat6 removed.asset.json

* Fix cs exceptions

* Update ExtensionHelper

* Rename Compat.php to Compat6.php

* Disable compat6 plugin for system tests

* Update administrator/language/en-GB/plg_behaviour_compat6.ini

Co-authored-by: Brian Teeman <brian@teeman.net>

* Update plugins/behaviour/compat6/compat6.xml

Co-authored-by: Brian Teeman <brian@teeman.net>

* Update plugins/behaviour/compat6/compat6.xml

Co-authored-by: Brian Teeman <brian@teeman.net>

* Update administrator/language/en-GB/plg_behaviour_compat6.ini

Co-authored-by: Brian Teeman <brian@teeman.net>

* Update administrator/language/en-GB/plg_behaviour_compat6.sys.ini

Co-authored-by: Brian Teeman <brian@teeman.net>

* Update plugins/behaviour/compat6/src/Extension/Compat6.php

* Restore class aliases from compat 5 plugin

* Relocate files from #44925

* Apply language string improvement from review suggestion

Co-authored-by: Brian Teeman <brian@teeman.net>

* Update plugins/behaviour/compat6/src/Extension/Compat6.php

Co-authored-by: Allon Moritz <allon.moritz@digital-peak.com>

* Update plugins/behaviour/compat6/compat6.xml

Co-authored-by: Allon Moritz <allon.moritz@digital-peak.com>

* Update installation/sql/mysql/base.sql

Co-authored-by: Richard Fath <richard67@users.noreply.github.com>

* Update installation/sql/postgresql/base.sql

Co-authored-by: Richard Fath <richard67@users.noreply.github.com>

* Fix wrong folder in uninstallation of compat (5) plugin.

---------

Co-authored-by: Richard Fath <richard67@users.noreply.github.com>
Co-authored-by: Richard Fath <richard.fath@t-online.de>
Co-authored-by: Brian Teeman <brian@teeman.net>
Co-authored-by: Benjamin Trenkle <bembelimen@users.noreply.github.com>
Co-authored-by: Allon Moritz <allon.moritz@digital-peak.com>

richard67 committed:
[6.0] Update deleted files and folders in script.php 2025-05-11 (#45462)

* Add deleted files and folders 2025-05-11

* Add deleted files from PR 45480 and update SQL scripts

* Add exceptions to deleted_file_check

* More exceptions for files deleted by plugin uninstallation

softforge committed:
Joomla! 6.0.0 Alpha 1

softforge committed:
Reset to dev

bembelimen committed:
Revert to dev

brianteeman committed:
[5.3] Installation missing string (#45520)

joomla-translation-bot committed:
[5.3] Translation Update (#45522)

brianteeman committed:
[5.3] Code comments (#45514)

Bodge-IT committed:
Merge remote-tracking branch 'upstream/5.4-dev' into upmerges/2025-05-28

Bodge-IT committed:
Test GPG signing verification

Bodge-IT committed:
Test GPG verification with matching email

Bodge-IT committed:
[6.0] Upmerges - 2025-05-28

Merge pull request #45534 from Bodge-IT/upmerges/2025-05-28

LadySolveig committed:
Update version numbers and migration guide links in github release notes layout (#45532)

Hackwar committed:
[6.0] Workflow: use generic AbstractApplication in type hint (#43155)

Dileepadari committed:
applied changed suggested in the review

Dileepadari committed:
applied changed suggested in the review

Dileepadari committed:
applied changed suggested in the review

# Conflicts:
#	administrator/language/en-GB/plg_workflow_category.ini

Dileepadari committed:
language string updated

Dileepadari committed:
language string updated

Dileepadari committed:
removed old files

Dileepadari committed:
removed old files

Dileepadari committed:
removed old files

joomla-translation-bot committed:
[5.3] Translation Update (#45537)

bembelimen commented on administrator/language/en-GB/plg_workflow_category.sys.ini:
```diff
@@ -0,0 +1,7 @@
+; License GNU General Public License version 2 or later; see LICENSE.txt
+; Note : All ini files need to be saved as UTF-8
+
+PLG_WORKFLOW_CATEGORY="Workflow - Category"
```
Please check if you need this language string also in the other file.

bembelimen commented on plugins/workflow/category/category.xml:
```diff
@@ -0,0 +1,35 @@
+	<config>
+		<fields name="params">
+			<fieldset name="basic">
+				<field
+					name="default_category"
+					type="category"
+					extension="com_content"
+					label="PLG_WORKFLOW_CATEGORY_LABEL"
+				/>
```
Thinking about this, I came to the conclusion, it makes no sense to have a default category. I mean, when in the transition no category is selected, no chance should happen at all (not all transitions neeed a category change)

bembelimen commented on plugins/workflow/category/forms/action.xml:
```diff
@@ -0,0 +1,16 @@
+                label="PLG_WORKFLOW_CATEGORY_FIELD_CATEGORY_LABEL"
+                description="PLG_WORKFLOW_CATEGORY_FIELD_CATEGORY_DESC"
+                extension="com_content"
+            />
```
Make sure, to have the posibility to not select a category at all.

Also workflow is more than "com_content", there could be 3rd party extension using categories + workflow, so there is the need for a dynamic way for the "extension".

bembelimen commented on plugins/workflow/category/src/Extension/Category.php:
```diff
@@ -0,0 +1,259 @@
+final class Category extends CMSPlugin implements SubscriberInterface
+{
+    use WorkflowPluginTrait;
+    use DatabaseAwareTrait;
```
If you don't use direct database calls, this trait is not needed

bembelimen commented on plugins/workflow/category/src/Extension/Category.php:
```diff
@@ -0,0 +1,259 @@
+            return;
+        }
+
+        if ($context === 'com_content.article') {
```
Here we should think about a way to not limit to com_content only (check the featuring plugin, how what options there are)

bembelimen commented on plugins/workflow/category/src/Extension/Category.php:
```diff
@@ -0,0 +1,259 @@
+    {
+        if (!$this->getApplication()->isClient('administrator')) {
+            return;
+        }
```
The following JS is added on each view in the backend, probably it can be shown only on the views where it's needed?

muhme committed:
[5.3] Fix build.php to create zstd packages on macOS (#45535)

bembelimen commented on plugins/workflow/category/src/Extension/Category.php:
```diff
@@ -0,0 +1,259 @@
+
+            $articleTable = $component->getMVCFactory()->createModel($modelName, $this->getApplication()->getName(), ['ignore_request' => true])
+                ->getTable();
+            if (!$articleTable->load($pk)) {
```
For coding an "else" in an "if-statement" is most of the time not best practise. A general rule of thumb is: no "else" but working with [early returns](https://dev.to/jpswade/return-early-12o5).

bembelimen commented on plugins/workflow/category/src/Extension/Category.php:
```diff
@@ -0,0 +1,259 @@
+                $articleTable->modified_by = $originalData->modified_by;
+
+                if (!$articleTable->store()) {
+                    $app->enqueueMessage(Text::sprintf('PLG_WORKFLOW_CATEGORY_ARTICLE_UPDATE_FAILED', $pk, $categoryId), 'error');
```
it's good to be that verbose, but you should not set all this enqueue message, otherwise the user is flooded. If something breaks, throw an exception and catch it somewhere later where you generate the minimum amount of messages (but as many as needed).

bembelimen commented on plugins/workflow/category/src/Extension/Category.php:
```diff
@@ -0,0 +1,259 @@
+                    $articleTable->catid = $categoryId;
+                }
+
+                $articleTable->modified = $originalData->modified;
+                $articleTable->modified_by = $originalData->modified_by;
```
I don't really know why this is needed?

bembelimen commented on plugins/workflow/category/src/Extension/Category.php:
```diff
@@ -0,0 +1,259 @@
+                }
+            });
+        ";
+        $this->getApplication()->getDocument()->addScriptDeclaration($js);
```
```addScriptDeclaration``` via document is deprecated, please use:
- the webasset manager
- a JS file (+ joomla.asset.json)

bembelimen commented on plugins/workflow/category/src/Extension/Category.php:
```diff
@@ -0,0 +1,259 @@
+    protected function disableCategoryField(Form $form, $data)
+    {
+        // Get the current category ID value
+        $catid = $form->getValue('catid');
```
not every category-field is called "catid", you can get the correct name like: https://github.com/joomla/joomla-cms/blob/5.3-dev/plugins/workflow/featuring/src/Extension/Featuring.php#L136

bembelimen commented on plugins/workflow/category/src/Extension/Category.php:
```diff
@@ -0,0 +1,259 @@
+        $form->setFieldAttribute('catid', 'readonly', 'true');
+        $form->setFieldAttribute('catid', 'value', $catid);
+
+        $js = "
```
Is the JS needed at all?

bembelimen commented on plugins/workflow/category/src/Extension/Category.php:
```diff
@@ -0,0 +1,259 @@
+     */
+    protected function isSupported($context)
+    {
+        return in_array($context, ['com_workflow.transition', 'com_content.article']);
```
Looks strange, but I have to test :-)

bembelimen commented on plugins/workflow/category/src/Extension/Category.php:
```diff
@@ -0,0 +1,259 @@
+            });
+        ";
+
+        $this->getApplication()->getDocument()->addScriptDeclaration($js);
```
The "problem" with JS is, that everybody could go in with the browser inspector and enables the JS again. So we need here a server-side stop, too.

richard67 committed:
[5.4] Add pre-update checks for backward compatibility plugins for update to Joomla 6 (#45493)

Adds 2 new checks to the "Required Settings" checks of the pre-update checker 

---------

brianteeman committed:
[5.3] Debug duplicate queries dark mode (#45527)

joomla-translation-bot committed:
[5.3] Translation Update (#45546)

* Language update

* Language update

* Language update

* Language update

* Language update

* Language update

* Language update

* Language update

* Language update

* Language update

* Language update

---------

Co-authored-by: Richard Fath <richard67@users.noreply.github.com>
Co-authored-by: Stefan Wendhausen <stefan.wendhausen@tec-promotion.de>

SniperSister committed:
[5.4] Improve autoupdate UX for local site (#45547)

tecpromotion committed:
[5.4] alpha sorting lines com_joomlaupdate.ini (#45541)

muhme committed:
Merge remote-tracking branch 'upstream/5.3-dev' into 5.4-upmerge-2025-06-03

richard67 committed:
Merge branch '5.4-dev' into 5.4-upmerge-2025-06-03

richard67 committed:
Merge pull request #45557 from muhme/5.4-upmerge-2025-06-03

[5.4] Upmerge 2025-06-03

Bodge-IT committed:
Merge remote-tracking branch 'upstream/5.4-dev' into upmerges/2025-06-04

Bodge-IT committed:
[6.0] Upmerges 2025-06-04

Bodge-IT committed:
[6.0] Upmerges - 2025-06-04

Merge pull request #45559 from Bodge-IT/upmerges/2025-06-04

brianteeman committed:
[5.4] remove trailing slash (#45566)

This pr removes the trailing slash from the hr element. It is not valid html and not required.

This is only in the testing sample data plugin so it is not distributed but someone might see it during development and mistakenly conclude that a trailing slash should be added

code review only

Signed-off-by: BrianTeeman <brian@teeman.net>

brianteeman committed:
[5.4] Deprecate language string (#45564)

This deprecation was missed with #43147

Signed-off-by: BrianTeeman <brian@teeman.net>

brianteeman committed:
[5.4] Tags error message (#45569)

As with most core components the tags component will display a different error message if the alias is not unique to when the alias is not unique and trashed.
This simple PR corrects the error message for tags. This will allow a future PR to provide more helpful error messages.
Signed-off-by: BrianTeeman <brian@teeman.net>

brianteeman committed:
[5.3] Installation error page (#45466)

brianteeman committed:
[5.4] Readme howto contribute (#45568)

Added 'How to submit code to the Joomla CMS using a PR' link to README.[md|txt]

---------

Signed-off-by: BrianTeeman <brian@teeman.net>

brianteeman committed:
[5.4] Tooltip width (#45571)

Make tooltips overflow-wrap to be readable on mobile devices

Signed-off-by: BrianTeeman <brian@teeman.net>

muhme committed:
Merge remote-tracking branch 'upstream/5.3-dev' into 5.4-upmerge-2025-06-10

richard67 committed:
Merge pull request #45591 from muhme/5.4-upmerge-2025-06-10

[5.4] Upmerge 2025-06-10

Bodge-IT committed:
Merge remote-tracking branch 'upstream/5.4-dev' into upmerges/2025-06-11

Hackwar committed:
[6.0] Installer: Remove dependency on Adapter class (#43792)

* Installer: Remove dependency on Adapter class

* Update ExtensionInstallCommand.php

* Fixing database injection

Bodge-IT committed:
Merge branch '6.0-dev' into upmerges/2025-06-11

Bodge-IT committed:
[6.0] Upmerges - 2025-06-11

Merge pull request #45593 from Bodge-IT/upmerges/2025-06-11

brianteeman committed:
[6.0] Reset stats on update to 6 (#45498)

The data gathered by the Joomla stats plugin is hopelessly useless. If a site owner chose the option to send the stats "just once" then we never receive any stats ever again. The option to send the stats "just once" might have been made with Joomla 3.5 back in 2013 and we have never received any data for that site since. The site admin might have changed numerous times since then and they wont even know about the stats plugin.

All of this makes the data we have not fit for purpose.

This PR resets the params for the plugin so that any site updating to joomla 6 will be asked again if they are willing to help the Joomla project by submitting their site stats.

_personally I would make this reset happen on every minor release (approx every 6 months) but to be less controversial this PR makes the reset happen on the next major release only.

brianteeman committed:
[6.x] Update strings (#45563)

Bump the version number in the com_joomlaupdate language strings

chmst committed:
[6.0] Accesslevel edit - simplify (#45588)

* Simplify access level edit
* deprecate language string
* Fix test
* add tab
* cs

brianteeman committed:
[6.x] Remove j5 guided tours (#45512)

* [6.x] Remove j5 guided tours

This PR removes the language files for the j5 whats new guided tours from NEW installs of j6 as these guided tours are not available on a clean install of j6
* remove images
* 54 image

laoneo committed:
Removes fetchExtensionCompatibility in UpdateController class (#45436)

AdarshSantoria committed:
[6.0] Enable support for multiple image extensions in Random Image module (#45200)

* Enable Multi extension random image
* fix style
* Update RandomImageHelper.php
* CS fix
---------

Hackwar committed:
[5.4] Remove calls to LegacyPropertyManagementTrait in User class context (#44905)

Removed calls to LegacyPropertyManagementTrait in User class context

Denitz committed:
[5.3] Optimize load of smart search plugins language (#45345)

* Update LanguageHelper.php

* Update LanguageHelper.php

---------

Co-authored-by: Quy Ton <quy@nomonkeybiz.com>

brianteeman committed:
remove string (#45565)

Hackwar committed:
[6.0] Updater: Remove Adapter classes from inheritance (#43793)

* Updater: Remove Adapter classes from inheritance
* Codestyle
* Update Updater.php
* Fixing database access
* Fixing database access - part 2
* Removing LegacyErrorHandlingTrait again from Updater and UpdateAdapter
* Removing unnecessary imports
* Removing duplicate getAdapter() call

muhme committed:
Merge remote-tracking branch 'upstream/5.3-dev' into 5.4-upmerge-2025-06-17

richard67 committed:
Merge pull request #45615 from muhme/5.4-upmerge-2025-06-17

[5.4] Upmerge 2025-06-17

laoneo committed:
[5.3] Disable opcache as it makes the SEF system test flaky (#45608)


Co-authored-by: Richard Fath <richard67@users.noreply.github.com>

Bodge-IT committed:
[6.0] Replace incorrectly removed  entries (#45618)

Bodge-IT committed:
Merge remote-tracking branch 'upstream/5.4-dev' into upmerges/2025-06-18

Bodge-IT committed:
[6.0] Upmerges - 2025-06-18

Merge pull request #45621 from Bodge-IT/upmerges/2025-06-18

laoneo committed:
Remove the dispatch event functions in model and view (#45431)

brianteeman committed:
[5.3] Remove Fields button for custom module (#45600)

chmst committed:
[5.3] Copy/Paste error in comment. (#45625)

brianteeman committed:
[5.3] Login Module show password (#45626)

richard67 committed:
Merge remote-tracking branch 'upstream/5.3-dev' into 5.4-dev-upmerge-2025-06-20

muhme committed:
Merge pull request #45632 from richard67/5.4-dev-upmerge-2025-06-20

[5.4] Upmerge 2025-06-20

Bodge-IT committed:
Merge remote-tracking branch 'upstream/5.4-dev' into upmerges/2025-06-20

Bodge-IT committed:
[6.0] Upmerges - 2025-06-20

Merge pull request #45633 from Bodge-IT/upmerges/2025-06-20

muhme committed:
Joomla! 5.4.0 Alpha 2

Bodge-IT committed:
Joomla! 6.0.0 Alpha 2

muhme committed:
Reset to dev

Bodge-IT committed:
Reset to dev

muhme committed:
[5.3] build.php Bug fix and stability improvements (#45622)

brianteeman committed:
[5.4] Media Delete Confirmation message (#45631)

* [5.4] Media Delete Confirmation message

This PR introduces a dedicated delete confirmation message for com_media instead of the global delete message.

The reason is to further clarify/warn that deleting a folder will delete all the contents of that folder. Unlike the generic message which just talks about deleting the selected item.

---------

Signed-off-by: BrianTeeman <brian@teeman.net>

laoneo committed:
[5.3] Add CI Github workflow (#45602)

HLeithner committed:
[5.3] Fix type for custom form validators (#45650)

joomla-translation-bot committed:
[5.3] Translation Update (#45643)

* Language update

* Language update

* Language update

* Language update

* Language update

* Language update

* Language update

* Language update

* Language update

* Language update

* Language update

* Language update

---------

Co-authored-by: Richard Fath <richard67@users.noreply.github.com>
Co-authored-by: Stefan Wendhausen <stefan.wendhausen@tec-promotion.de>

joomlaweby committed:
[5.4] Convert mod_random_image to service provider (#45648)

* mod_random_image converted to service provider

Hackwar committed:
[5.4] Error handling: Adding new shouldUseException() (#44098)

To make an easy transition from the legacy error handling with "getError" and "setError" methods, this PR introduces new methods "shouldUseExceptions" and "setUseExceptions" to the "LegacyErrorHandlingTrait". The latter allows to set a flag if exceptions should be used or not, and the former allows to check if that flag is set or not. The plan is to add this to 5.4 and then to remove all of this from 7.0 again.

joomla-translation-bot committed:
[5.3] Translation Update (#45663)

* Language update

* Language update

* Language update

* Language update

* Language update

* Language update

* Language update

* Language update

* Language update

* Language update

* Language update

* Language update

* Language update

---------

Co-authored-by: Richard Fath <richard67@users.noreply.github.com>
Co-authored-by: Stefan Wendhausen <stefan.wendhausen@tec-promotion.de>

bembelimen committed:
Joomla! 5.3.2 RC 1

bembelimen committed:
Revert to dev

bembelimen committed:
Merge branch '5.3-dev' of github.com:joomla/joomla-cms into 5.3-dev

bembelimen committed:
Joomla! 5.3.2

bembelimen committed:
Reset to dev

Hackwar committed:
[5.4] Deprecating UCM code (#44910)

The UCM system was added in Joomla 3.2.0 and already at that time had major flaws. In 12 years of development, it hasn't been extended in any way and instead we've rather reduced it instead. (The content versioning was a part of it and has been made standalone in 4.0.) I think it can be said that the concept (which barely exists to start with) has failed and we should remove the remnants of it.
This PR deprecates all UCM classes that I could find. It also removes the code where we are using the UCM classes in a functional way. For the moment it keeps it, where it might be used by third party code to keep it backwards compatible. This means that we keep this code in the batch methods, where $this->contentType and $this->type is set with this code, even though neither of the two attributes are used anywhere. We could remove this in 6.0 and drop everything in 7.0 finally.

obuisard committed:
[5.4] Automated Update Quickicon language keys and icon (#45669)

* Changed icon and language keys to remove 'health' notions

muhme committed:
Merge remote-tracking branch 'upstream/5.3-dev' into 5.4-upmerge-2025-07-01

- Merge 5.3-dev PRs into 5.4-dev.
- 10 files with merge conflicts, but only version and date/time. All incoming changes were ignored, and the current version was kept.

richard67 committed:
Merge pull request #45671 from muhme/5.4-upmerge-2025-07-01

[5.4] Upmerge 2025-07-01

SniperSister committed:
[5.4] Fix handling of responses from autoupdate server (#45673)

Fix response handling for undefined array

---------

Co-authored-by: @Ruud68

Bodge-IT committed:
Merge remote-tracking branch 'upstream/5.4-dev' into upmerges/2025-07-02

Bodge-IT committed:
[6.0] Upmerges 2025-07-02

Merge pull request #45677 from Bodge-IT/upmerges/2025-07-02

Hackwar committed:
[5.4] Removing usage of deprecated constants (#45249)

* Removing usage of
  * JPATH_COMPONENT_SITE
  * JPATH_COMPONENT_ADMINISTRATOR
  * JPATH_COMPONENT
* Reverting changes to UpdateModel

---------

Co-authored-by: Quy Ton <quy@nomonkeybiz.com>
Co-authored-by: Robert Deutz <rdeutz@googlemail.com>

brianteeman committed:
[5.4] Blog Sample Data Module (#45692)

Fixes the "Site Information" dashboard module (added by blog sample data) to use the correct heading level (<h2> instead of <h3>) and sets the Bootstrap width to 12pt for consistency with other admin modules.

Bodge-IT committed:
Merge remote-tracking branch 'upstream/5.4-dev' into upmerges/2025-07-09

Bodge-IT committed:
[6.0] Upmerges - 2025-07-09

Merge pull request #45709 from Bodge-IT/upmerges/2025-07-09

Denitz committed:
Also count subfolder files in the ‘Maintenance: Delete cache’ (#45686)

chmst committed:
[5.4] Send Automated Update Notifications to all super users (#45685)

Removing email input field

bembelimen committed:
[5.4] Fix registration error notification (#45656)

On email error the user is no longer redirected to the registration form

joomlaweby committed:
[5.4] Convert mod_menu to service provider (#45687)

brianteeman committed:
[5.4] configurated (#45706)

Corrects the spelling of the 2 language keys *CONFIGURATED*.

brianteeman committed:
[5.4] GitHub action typos (#45705)

Added GitHub action 'typos' and configures exceptions (exclusions) from the check

heelc29 committed:
[5.4] Local Filesystem plugin use SubscriberInterface (#45703)

joomla-translation-bot committed:
[5.3] Translation Update (#45675)

rdeutz committed:
[5.3] Phpstan baseline 5 3 (#45666)

brianteeman committed:
[5.3] Whitelist (#45723)

brianteeman committed:
[5.4] Deprecate language string Mis_typed (#45727)

To fix the spelling of a key and to be able to remove an exception in typos.toml this PR deprecates a mis-spelled language key and replaces it in the error.php templates with the correct string

typos.toml will be updated in a subsequent pr

Signed-off-by: BrianTeeman <brian@teeman.net>

travisrisner committed:
swap hardcoded bg-warning text color to black on installation (#45717)

brianteeman committed:
[5.4] Deprecate language string (#45725)

* [5.4] Deprecate language string

To fix the spelling of a string and to be able to remove an exception in typos.toml this PR deprecates a mis-spelled language string and replaces it in the xml with the correct string

Signed-off-by: BrianTeeman <brian@teeman.net>

brianteeman committed:
[5.4] typos.toml (#45729)

Sorts and explains the exceptions used by the typos action, also removes two that were fixed upstream.

heelc29 committed:
[5.3] phpstan-baseline.neon should not be distributed (#45730)

Exclude the file `phpstan-baseline.neon` from the packages to be built.

richard67 committed:
Merge branch '5.3-dev' into 5.4-dev-upmerge-2025-07-15

richard67 committed:
Adapt phpstan baseline to 5.4-dev

muhme committed:
Merge pull request #45732 from richard67/5.4-dev-upmerge-2025-07-15

[5.4] Upmerge 2025-07-15

richard67 committed:
[5.4] Add deleted files from PRs #45648 and #45687 (#45718)

Bodge-IT committed:
Merge remote-tracking branch 'upstream/5.4-dev' into upmerges/2025-07-16

Bodge-IT committed:
Update CssMenu.php

Remove obsolete use Joomla\CMS\Factory, not highlighted during merge.

Bodge-IT committed:
[6.0] Upmerges - 2025-07-16

Merge pull request #45734 from Bodge-IT/upmerges/2025-07-16

Drone failure expected and overridden.

rdeutz committed:
[6.0] Phpstan baseline (#45655)

* use baseline and remove drone ignore failue
* trigger build
* Update phpstan-baseline.neon

---------

Co-authored-by: Robert Deutz <rd@rdbs.de>
Co-authored-by: Richard Fath <richard.fath@t-online.de>

brianteeman committed:
[6.0] Module Tag Field (#45752)

* [6.0] Module Tag Field

Adds search to the available options

* Update ModuletagField.php

alpha

---------

Co-authored-by: Quy Ton <quy@nomonkeybiz.com>

chmst committed:
Tag search for mod-finder (#45744)

Improve a11y for mod finder using the search tag and remove role="search".

muhme committed:
Joomla! 5.4.0 Alpha 3

muhme committed:
Reset to dev

softforge committed:
Release 6.0.0 Alpha 3

softforge committed:
Revert to dev

muhme committed:
Reset to dev with alpha4-dev

SniperSister committed:
[5.4] Fix auto-registration for autoupdates for new installs (#45754)

Auto-Update registration state is correctly stored in the config.

brianteeman committed:
[5.4] Stats plugin accessibility (#45736)

Signed-off-by: BrianTeeman <brian@teeman.net>

brianteeman committed:
[5.4] Remove outdated reference to fof in comments (#45740)

Remove the reference to fof and remove the execption in the typos action for "fof"

Bodge-IT committed:
Merge remote-tracking branch 'upstream/5.4-dev' into upmerges/2025-07-23

Bodge-IT committed:
[6.0] Upmerges - 2025-07-23

Merge pull request #45764 from Bodge-IT/upmerges/2025-07-23

MacJoom committed:
[6.0] Remove call to 'toPunycode' which throws error for files with '_' (#45627)

* Remove call to 'toPunycode' which throws error for files with underscores

* Remove library

---------

Co-authored-by: Quy Ton <quy@nomonkeybiz.com>

obuisard committed:
Specify the tour is for article categories (#45755)

brianteeman committed:
[6.x] Remove deprecated strings (#45738)

Remove three strings marked for removal in 6.0

Remove exceptions from typos config

code review

brianteeman committed:
[6.x] Remove deprecated blacklist (#45722)

Removes code deprecated in favour of forbiddenlist

Original PR with explanation of the deprecation #32030

RickR2H committed:
[6.0] Add custom classes to the class dropdown in the image dialog in TinyMCE (#45676)

* Added option to define image classes

* Making phpcs happy

raj20889 committed:
Fix: Updated Joomla content form with proper tab handling, field rendering, and conditional visibility for images and metadata fields. (#45044)

Dileepadari committed:
changes done for improvements

Dileepadari committed:
Merge branch 'category_transition_test' of https://github.com/Dileepadari/joomla-cms into category_transition_test
